src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -113,118 +113,86 @@ public async Task<Solution> CreateSolutionAsync(Checksum newSolutionChecksum, Ca
             }
 
             private async Task<Solution> UpdateProjectsAsync(
-                Solution solution, SolutionStateChecksums oldSolutionChecksums, SolutionStateChecksums newSolutionChecksums, CancellationToken cancellationToken)
+                Solution solution, SolutionStateChecksums newSolutionChecksums, CancellationToken cancellationToken)
```

Comment:
this helper was pretty significantly updated.  the old logic was pretty confusing IMO, and didn't handle well teh new case where we might want to only be updating a subportion of a solution.  i recommend just looking at the 'after' here.